### ISSUE TITLE
Updates for install json

### ISSUE THREAD

**Issue by the-sane** [starcitizen-lug/lug-helper#52]
Courtesy of ngh in our discord:

Add 
`"system_winetricks": false,` 
to the wine part of the install json

While at it, might as well add 
`"dxvk_nvapi": false` 
so Nvidia users stop getting confused by that outdated driver warning

**Comment by the-sane** [comment]
If we do this, I should make some adjustments to the winetricks detector in our preflight check. We should probably find a way to check the winetricks version Lutris is using, or maybe just check the Lutris version itself?

**Comment by Termuellinator** [comment]
Is winetricks really bundled again in lutris?
I only found that it has been removed in 0.5.5 but nothing about it being bundled again in later versions?

**Comment by the-sane** [comment]
Checked with ngh in discord and it looks like it's bundled again based on the info in https://lutris.net/api/runtime. So it's probably worth changing these things

**Comment by the-sane** [comment]
I adjusted the Helper's preflight check so that it looks for an updated Lutris version and changed the messaging on the winetricks check to mention the lutris setting. I also removed install winetricks as a step in our quick start guide.

@beniwtv Everything should be ready for us to update the install json whenever you have a chance to do that.

**Comment by beniwtv** [comment]
Will do it ASAP.

**Comment by the-sane** [comment]
Looks like the GE runner we currently bundle isn't working for a lot of people even if they set the new arguments/CLI mode. GE's latest runner v35 sounds like it sorts most people out. Maybe we should be switching to that?

I also saw some mention that `"libglesv2": "builtin"` might not be needed anymore but I'm waiting for more details from the folks who said that.

**Comment by the-sane** [comment]
Some people are also saying the new launcher is unable to create this directory:
`"Error installing at C:\Program Files\Roberts Space Industries\StarCitizen\LIVE: Error: ERR_INTERNAL: CreateFile failed. Path not found.\r\n"`

I might have to update the Helper to create it before sending the install json to Lutris.

**Comment by the-sane** [comment]
On second thought, thinking through that, it would require a significant restructuring of the way the Helper currently tries to automatically detect those directories. We'd end up in a weird situation where it asks for the wine prefix and game install path before those things even exist on the system. Since they don't exist, the user would have to manually type them somewhere instead of selecting them from a file picker dialog as they do now. That's very messy and I don't particularly want to do that.

@beniwtv Is there any way we can fix this in the installer?

**Comment by the-sane** [comment]
> I also saw some mention that `"libglesv2": "builtin"` might not be needed anymore but I'm waiting for more details from the folks who said that.

Answer: It's no longer needed for those using DXVK 2.1

**Comment by the-sane** [comment]
For creating a directory during installation, I went looking through the lutris install script documentation. There are commands for moving and copying files, but not necessarily creating a directory structure. A dirty way to do it might be to create a blank text file at that path. Otherwise, create_prefix creates an empty prefix and then there's a `wineexec` command that runs a windows executable. Maybe that can run mkdir if it exists in wine?

**Comment by the-sane** [comment]
Updated list of all the changes to be made:

Add:
- `"system_winetricks": false`
- `"dxvk_nvapi": false`
- Some way of creating the directory structure for `C:\Program Files\Roberts Space Industries\StarCitizen\LIVE` and also `PTU`

Change:
- Update RSI installer version
- Update GE runner to the latest v35

Remove:
Maybe later, once most people migrate to >=DXVK 2.1
- `"libglesv2": "builtin"`

**Comment by beniwtv** [comment]
I'll have to check the thing with the path; sounds strange to me. Could be a weird file system, or people having tried as sudo when something didn't work.

**Comment by the-sane** [comment]
> Can we set Prefer system libraries to Off for the duration of the installation and then reset it to default afterwards? Some people reported their install hangs without this set globally first.

Someone else just had to toggle it on to make their install work. So maybe there's nothing we can do about this automatically right now if it's not a one-size-fits-all solution.

**Comment by the-sane** [comment]
Changes suggested by ProfDecoy on our Discord to solve the directory problem. Also updates the installer and runner:

```
$ git diff
diff --git a/lug-lutris-install.json b/lug-lutris-install.json
index cebf793..c9229b6 100644
--- a/lug-lutris-install.json
+++ b/lug-lutris-install.json
@@ -32,7 +32,7 @@
       "script": {
         "files": [
           {
-            "client": "https://install.robertsspaceindustries.com/star-citizen/RSI-Setup-1.5.5.exe"
+            "client": "https://install.robertsspaceindustries.com/star-citizen/RSI-Setup-1.6.2.exe"
           }
         ],
         "game": {
@@ -67,6 +67,11 @@
               "name": "wineexec",
               "prefix": "$GAMEDIR"
             }
+          },
+          {
+            "execute": {
+              "command": "mkdir -p $GAMEDIR'/drive_c/Program Files/Roberts Space Industries/StarCitizen/'{LIVE,PTU}"
+            }
           }
         ],
         "system": {
@@ -85,7 +90,7 @@
             "nvapi,nvapi64": "disabled",
             "libglesv2": "builtin"
           },
-          "version": "lutris-GE-Proton7-15-SC-x86_64"
+          "version": "lutris-GE-Proton7-35-x86_64"
         }
       }
     }
```

Changes left to make:

`"system_winetricks": false`
`"dxvk_nvapi": false`

Also something for us to think about is the EAC workaround. I've switched to mangling the EAC json file on my machine instead of applying it globally. We can set a pre-launch script for this and then mark the json file as immutable so launcher updates don't undo the change. See my instructions on our [tips and tricks](https://github.com/starcitizen-lug/information-howtos/wiki/Tips-and-Tricks#easy-anti-cheat-workaround) page. Is this something worth changing? Or do we keep things as-is with the hosts workaround?

Benefits:
- Only applies the EAC workaround to Star Citizen. Won't mess up other games.

Disadvantages:
- May not be entirely future-proof if CIG changes things? ie, if the EAC json file actually needs some update from CIG but we have it immutable.

If we go that route, I'd say we should put all this stuff into a separate script file that we host alongside the Helper, otherwise it becomes one very long and unwieldy shell script command. We originally discussed doing that, but decided at the time that it was simpler to put the one command right in the json file.

**Comment by the-sane** [comment]
Updates made in https://github.com/starcitizen-lug/lug-helper/commit/636c55cf812be9787892acdd8edc5149c5372365

I decided to stick with the hosts file workaround based on discussions over in discord. The json approach is limited in that it may fail after the first install or launcher updates. That's not idea for new users.

I decided not to disable dxvk_nvapi yet because I can't find information on whether this has consequences for amd users or if it's really needed for every nvidia user.
